DataDog/dd-trace-rb
# set 128-bit trace_id to true by default

**What does this PR do?**
Changes the default configuration for 128-bit trace_id generation to `true`. 

**Motivation:**
Part of Datadog's migration to W3C/128-bit tracers for OTEL compatibility. 

**Additional Notes:**
When using 128-bit IDs internally, we propagate the 64-bit version in `x-datadog-tags`, which we do not do for 64-bit IDs. Some unit tests had to be updated to reflect this. 

**How to test the change?**
Since this changes default behavior you just have to use the tracer with default settings to see the change. 

**For Datadog employees:**
credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

Unsure? Have a question? Request a review!
